node/src/database/db_initializer.rs

```diff
@@ -357,51 +331,61 @@ impl DbInitializerReal {
         }
     }
 
-    fn extract_configurations(&self, conn: &Connection) -> HashMap<String, (Option<String>, bool)> {
-        let mut stmt = conn
-            .prepare("select name, value, encrypted from config")
-            .unwrap();
-        let query_result = stmt.query_map([], |row| {
-            Ok((
-                row.get(0),
-                row.get(1),
-                row.get(2).map(|encrypted: i64| encrypted > 0),
-            ))
-        });
-        match query_result {
-            Ok(rows) => rows,
-            Err(e) => panic!("Error retrieving configuration: {}", e),
-        }
-        .map(|row| match row {
-            Ok((Ok(name), Ok(value), Ok(encrypted))) => (name, (Some(value), encrypted)),
-            Ok((Ok(name), Err(InvalidColumnType(1, _, _)), Ok(encrypted))) => {
-                (name, (None, encrypted))
+    fn check_migrations_and_give_conn(
+        &self,
+        conn: Connection,
+        init_config: DbInitializationConfig,
+        db_file_path: &Path,
+        target_version: usize,
+        flags: OpenFlags,
+    ) -> Result<Box<dyn ConnectionWrapper>, InitializationError> {
+        let str_sv = Self::read_current_schema_version(&conn)?;
+        match (Self::is_migration_required(&str_sv)?, init_config.init_mode) {
+            (None, _) => Ok(Box::new(ConnectionWrapperReal::new(conn))),
+            (Some(_), InitializationMode::CreationBannedMigrationPanics) => {
+                panic!("Broken code: Migrating database at inappropriate place")
             }
-            e => panic!("Error retrieving configuration: {:?}", e),
-        })
-        .collect::<HashMap<String, (Option<String>, bool)>>()
+            (
+                Some(mismatched_version),
+                InitializationMode::CreationAndMigration { external_data },
+            ) => {
+                let migrator = Box::new(DbMigratorReal::new(external_data));
+                self.migrate_and_return_connection(
+                    conn,
+                    mismatched_version,
+                    target_version,
+                    db_file_path,
+                    flags,
+                    migrator,
+                )
+            }
+            (Some(_), InitializationMode::CreationBannedMigrationSuppressed) => {
+                Ok(Box::new(ConnectionWrapperReal::new(conn)))
+            }
+            (Some(_), InitializationMode::CreationBannedMigrationThrowsErr) => {
```

Comment:
"Throws" isn't a good term to use here, because it implies exceptions, and Rust doesn't use exceptions. Maybe "ReturnsErr" or "RaisesErr" or "Complains?"